src/NodeTypeResolver/PHPStan/Scope/NodeVisitor/PhpVersionConditionNodeVisitor.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\PHPStan\Scope\NodeVisitor;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\Ternary;
+use PhpParser\Node\Stmt\If_;
+use PhpParser\NodeVisitorAbstract;
+use Rector\Contract\PhpParser\DecoratingNodeVisitorInterface;
+use Rector\DeadCode\ConditionResolver;
+use Rector\DeadCode\ValueObject\VersionCompareCondition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\PhpParser\NodeTraverser\SimpleNodeTraverser;
+
+final class PhpVersionConditionNodeVisitor extends NodeVisitorAbstract implements DecoratingNodeVisitorInterface
```

Comment:
instead of this, you might be interessted in `$scope->getPhpVersion()` which returns a version which can be narrowed by composer.json constraints and/or `if(PHP_VERSION_ID > xy)` check etc.

see also https://staabm.github.io/2024/11/14/phpstan-php-version-narrowing.html